### PR TITLE
[tictactoe] draw bug fix and tests

### PR DESCRIPTION
**Pull request by LuZhang-Lou**
We have a couple of bugs in the code (for both `TicTacToe` and `SharedTicTacToe`) that prevents draw to be identified. 
This PR adds fixes and tests for draw scenario.

Details will be commented inline

### CODE REVIEW

File: sui_programmability/examples/games/sources/SharedTicTacToe.move [hunk 1]
```diff
@@ -150,7 +149,7 @@ module Games::SharedTicTacToe {
         check_for_winner(game, 2, 0, 1, 1, 0, 2);
 
         // Check if we have a draw
-        if (game.game_status == IN_PROGRESS && game.cur_turn == 9) {
+        if (game.game_status == IN_PROGRESS && game.cur_turn == FINAL_TURN) {
```
Comment (LuZhang-Lou) [inline sui_programmability/examples/games/sources/SharedTicTacToe.move:152]: FINAL_TURN = 8, cur_turn starts from 0

File: sui_programmability/examples/games/sources/TicTacToe.move [hunk 2]
```diff
@@ -231,7 +232,7 @@ module Games::TicTacToe {
         check_for_winner(game, 2, 0, 1, 1, 0, 2);
 
         // Check if we have a draw
-        if (game.game_status != IN_PROGRESS && game.cur_turn == 9) {
+        if (game.game_status == IN_PROGRESS && game.cur_turn == FINAL_TURN) {
```
Comment (LuZhang-Lou) [inline sui_programmability/examples/games/sources/TicTacToe.move:235]: same thing here, also `!=` -> `==`

File: sui_programmability/examples/games/sources/SharedTicTacToe.move [hunk 3]
```diff
@@ -93,9 +92,9 @@ module Games::SharedTicTacToe {
         let cell = Vector::borrow_mut(Vector::borrow_mut(&mut game.gameboard, (row as u64)), (col as u64));
         assert!(*cell == MARK_EMPTY, ECELL_OCCUPIED);
 
-        *cell = game.cur_turn;
+        *cell = game.cur_turn % 2;
         update_winner(game);
-        game.cur_turn = if (game.cur_turn == MARK_X) MARK_O else MARK_X;
+        game.cur_turn = game.cur_turn + 1;
```
Comment (LuZhang-Lou) [inline sui_programmability/examples/games/sources/SharedTicTacToe.move:97]: this implementation of rotating `cur_turn` does not help us track the total number of turns that already took place, which is used to check draws. I'm updating it to increments as in the `TicTacToe` version

File: sui_programmability/examples/games/tests/SharedTicTacToeTests.move [hunk 4]
```diff
@@ -68,13 +69,125 @@ module Games::SharedTicTacToeTests {
...
+        assert!(TestScenario::can_remove_object<Trophy>(scenario), 1);
+
+        TestScenario::next_tx(scenario, &player_o);
+        // O has no Trophy
+        assert!(!TestScenario::can_remove_object<Trophy>(scenario), 2);
+    }
+
+
+    #[test]
+    fun play_tictactoe_draw() {
```
Comment (LuZhang-Lou) [inline sui_programmability/examples/games/tests/SharedTicTacToeTests.move:84]: add test case for draw

File: sui_programmability/examples/games/tests/TicTacToeTests.move [hunk 5]
```diff
@@ -69,9 +71,124 @@ module Games::TicTacToeTests {
...
+        assert!(TestScenario::can_remove_object<Trophy>(scenario), 1);
+
+        TestScenario::next_tx(scenario, &player_o);
+        // O has no Trophy
+        assert!(!TestScenario::can_remove_object<Trophy>(scenario), 1);
+    }
+
+
+    #[test]
+    fun play_tictactoe_draw() {
```
Comment (LuZhang-Lou) [inline sui_programmability/examples/games/tests/TicTacToeTests.move:87]: same

**Review by lxfind — APPROVED** [review]
Good catch! Thank you!

**Pull request merged**